### PR TITLE
Fix cloud UA unmarshal

### PR DESCRIPTION
**Pull request by edaniels**
A fix of a similar vein. Cloud login was not working on dev due to us assuming the keys of the model instead of using the codec when doing `unmarshallUserAuth`.

### CODE REVIEW

File: src/auth/index.js [hunk 1]
```diff
@@ -274,4 +274,65 @@ export default class Auth {
     this.storage.remove(authCommon.IMPERSONATION_USER_KEY);
     this.storage.remove(authCommon.IMPERSONATION_REAL_USER_AUTH_KEY);
   }
+
+  parseRedirectFragment(fragment, ourState) {
```
Comment (edaniels) [inline src/auth/index.js]: I don't love this here, but we are breaking dev and therefore breaking any cloud testing on dev (Harry noticed this)

File: src/auth/index.js [hunk 2]
```diff
@@ -274,4 +274,65 @@ export default class Auth {
...
+    // After being redirected from oauth, the URL will look like:
+    // https://todo.examples.stitch.mongodb.com/#_stitch_state=...&_stitch_ua=...
+    // This function parses out stitch-specific tokens from the fragment and
+    // builds an object describing the result.
+    const vars = fragment.split('&');
+    const result = { ua: null, found: false, stateValid: false, lastError: null };
+    let shouldBreak = false;
+    for (let i = 0; i < vars.length; ++i) {
```
Comment (mburdette) [inline src/auth/index.js]: [nit] you could use `vars.forEach(var =>`

File: src/auth/index.js [hunk 3]
```diff
@@ -274,4 +274,65 @@ export default class Auth {
...
+      const pairParts = vars[i].split('=');
+      const pairKey = decodeURIComponent(pairParts[0]);
+      switch (pairKey) {
+      case authCommon.STITCH_ERROR_KEY:
+        result.lastError = decodeURIComponent(pairParts[1]);
```
Comment (mburdette) [inline src/auth/index.js]: [semi-nit] I think i would prefer making another const for this `pairPart` just to be more descriptive.

File: src/auth/index.js [hunk 4]
```diff
@@ -274,4 +274,65 @@ export default class Auth {
...
+        result.found = true;
+        let theirState = decodeURIComponent(pairParts[1]);
+        if (ourState && ourState === theirState) {
+          result.stateValid = true;
+        }
+        continue;
+      default: continue;
+      }
+
+      if (shouldBreak) {
```
Comment (mburdette) [inline src/auth/index.js]: you could just add this to the for loop conditional i think

File: src/auth/index.js [hunk 5]
```diff
@@ -274,4 +274,65 @@ export default class Auth {
...
+        break;
+      }
+    }
+
+    return result;
+  }
+
+  unmarshallUserAuth(data) {
+    let parts = data.split('$');
+    if (parts.length !== 4) {
```
Comment (mburdette) [inline src/auth/index.js]: [nit] might want to put this `4` in a const somewhere higher up in the file, although not super important

**Pull request merged**